network_test.go

```diff
@@ -0,0 +1,410 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package firecracker
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"path/filepath"
+	"sync"
+	"testing"
+	"time"
+
+	models "github.com/firecracker-microvm/firecracker-go-sdk/client/models"
+	"github.com/firecracker-microvm/firecracker-go-sdk/fctesting"
+	"github.com/sparrc/go-ping"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	mockMacAddrString = "00:11:22:33:44:55"
+	tapName           = "tap0"
+
+	cniNetworkName = "phony-network"
+	mockNetNSPath  = "/my/phony/netns"
+
+	kernelArgsNoIP   = parseKernelArgs("foo=bar this=phony")
+	kernelArgsWithIP = parseKernelArgs("foo=bar this=phony ip=whatevz")
+
+	validIPConfiguration = &IPConfiguration{
+		IPAddr: net.IPNet{
+			IP:   net.IPv4(10, 0, 0, 2),
+			Mask: net.IPv4Mask(255, 255, 255, 0),
+		},
+		Gateway:     net.IPv4(10, 0, 0, 1),
+		Nameservers: []string{"1.1.1.1", "8.8.8.8"},
```

Comment:
Is there a particular reason you're using public nameservers as opposed to other [RFC 1918](https://tools.ietf.org/html/rfc1918) addresses?

Reply:
Or [IPv4 Address Blocks Reserved for Documentation](https://tools.ietf.org/html/rfc5737)?

In the meantime, I can quickly recognize 8.8.8.8 and 1.1.1.1 as DNS servers, which is good, while `Nameservers:` pretty much explain that though.

Reply:
These values are not used in e2e tests or any time a real machine is spun up, they are just used for validation unit tests.

However, I see the point that it's possible for someone to inadvertently use them in actual e2e tests in the future, so it's probably best to change them. I'll update to IPs from the range @kzys linked to, which seems safest.